schedview/app/scheduler_dashboard/scheduler_dashboard.py

```diff
@@ -487,14 +543,57 @@ def read_scheduler(self):
 
             return False
 
+    def update_conditions(self):
+        """Update Conditions object.
+
+        Returns
+        -------
+        success : 'bool'
+            Record of success of Conditions update.
+        """
+        if self._conditions is None:
+            self._debugging_message = "Cannot update Conditions object as no Conditions object is loaded."
+
+            return False
+
+        try:
+            self._debugging_message = "Starting to update Conditions object."
+            pn.state.notifications.info("Updating Conditions object...", duration=0)
+
+            # self._conditions.mjd = self._mjd
+
+            # Use instance of ModelObservatory until Conditions
+            # setting bug is fixed.
+            self._model_observatory = ModelObservatory(
```

Comment:
If `self._model_observatory` already exists and has the correct nside, you should be able to just reuse it and avoid the significant time it takes to create a new instance.